pkg/k8sutil/k8sutil.go

```diff
@@ -175,20 +176,30 @@ func UpdateStatefulSet(ctx context.Context, sstClient clientappsv1.StatefulSetIn
 	return nil
 }
 
-// UpdateSecret merges metadata of existing Secret with new one and updates it.
-func UpdateSecret(ctx context.Context, secretClient clientv1.SecretInterface, secret *v1.Secret) error {
-	existingSecret, err := secretClient.Get(ctx, secret.Name, metav1.GetOptions{})
-	if err != nil {
-		return errors.Wrap(err, "getting secret object failed")
-	}
-
-	mergeMetadata(&secret.ObjectMeta, existingSecret.ObjectMeta)
-
-	_, err = secretClient.Update(ctx, secret, metav1.UpdateOptions{})
+// CreateOrUpdateSecret merges metadata of existing Secret with new one and updates it.
+func CreateOrUpdateSecret(ctx context.Context, secretClient clientv1.SecretInterface, desired *v1.Secret) error {
+	existingSecret, err := secretClient.Get(ctx, desired.Name, metav1.GetOptions{})
 	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(
+				err,
+				"failed to check whether secret %q in namespace %q already exists",
+				desired.Name,
+				desired.Namespace,
+			)
+		}
+		_, err = secretClient.Create(ctx, desired, metav1.CreateOptions{})
 		return err
 	}
-
+	mutated := existingSecret.DeepCopyObject().(*v1.Secret)
+	mergeMetadata(&desired.ObjectMeta, mutated.ObjectMeta)
+	// diff and whether update
+	if !apiequality.Semantic.DeepEqual(existingSecret, desired) {
+		_, err = secretClient.Update(ctx, desired, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Lets add secret name/namespace. Also a bit nicer to read IMHO (less nested levels).

```suggestion
	if apiequality.Semantic.DeepEqual(existingSecret, desired) {
                return nil
        }
	
	if _, err = secretClient.Update(ctx, desired, metav1.UpdateOptions{}); err != nil {
		return errors.Wrapf(
			err,
			"failed to update secret %q in namespace %q",
			desired.Name,
			desired.Namespace,
		)
	}
	
	return nil
```